source/Camera/Basic/C++Halcon/CaptureViaGenICam/CaptureViaGenICam.cpp

```diff
@@ -4,43 +4,35 @@
 #include <chrono>
 #include <iostream>
 
-
-void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string &filename)
+void savePointCloud(const HalconCpp::HObjectModel3D &model, const std::string &fileName)
 {
-	model.WriteObjectModel3d(HalconCpp::HString{ "ply" },
-		HalconCpp::HString{ filename.c_str() },
-		HalconCpp::HString{ "invert_normals" },
-		HalconCpp::HString{ "false" });
+	model.WriteObjectModel3d(HalconCpp::HString{"ply"},
+							 HalconCpp::HString{fileName.c_str()},
+							 HalconCpp::HString{"invert_normals"},
+							 HalconCpp::HString{"false"});
 }
 
-
-void setColorsInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const HalconCpp::HImage &RGB)
+void setColorsInObjectModel3D(HalconCpp::HObjectModel3D &objectModel3D, const HalconCpp::HImage &RGB, const HalconCpp::HImage &ZReduced)
 {
-	const HalconCpp::HRegion domain = RGB.AccessChannel(1).GetDomain();
+	const HalconCpp::HRegion domain = ZReduced.GetDomain();
 	HalconCpp::HTuple rows, cols;
 	domain.GetRegionPoints(&rows, &cols);
 
-	const HalconCpp::HTuple Rtup = RGB.AccessChannel(1).GetGrayval(rows, cols);
-	const HalconCpp::HTuple Gtup = RGB.AccessChannel(2).GetGrayval(rows, cols);
-	const HalconCpp::HTuple Btup = RGB.AccessChannel(3).GetGrayval(rows, cols);
-
-	std::cout << "Adding RGB to ObjectModel3D" << std::endl;
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", Rtup);
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", Gtup);
-	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", Btup);
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("red"), "points", RGB.AccessChannel(1).GetGrayval(rows, cols));
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("green"), "points", RGB.AccessChannel(2).GetGrayval(rows, cols));
+	objectModel3D.SetObjectModel3dAttribMod(HalconCpp::HTuple("blue"), "points", RGB.AccessChannel(3).GetGrayval(rows, cols));
 }
 
-
 HalconCpp::HString getFirstAvailableZividDevice(const HalconCpp::HTuple &devices)
 {
 	auto zividDevices = devices.TupleRegexpSelect("Zivid");
```

Comment:
I think it makes sense to add here the following lines that are currently in main? (Main would look nicer then). Do you see any reason not to do it?

`auto availableDevices = HalconCpp::HTuple();
		auto information = HalconCpp::HTuple();
		HalconCpp::InfoFramegrabber("GenICamTL", "device", &information, &availableDevices);`

Reply:
I agree, moved all inside getFirstAvailableZividDevice function. Main looks nicer now!